pydantic/_internal/_typing_extra.py

```diff
@@ -1,192 +1,414 @@
+    ```python test="skip" lint="skip"
+    is_new_type(NewType('MyInt', int))
+    #> True
+    ```
     """
-    if not is_literal_type(type_):
-        return [type_]
+    if sys.version_info < (3, 10):
+        # On Python < 3.10, `typing.NewType` is a function
+        return hasattr(tp, '__supertype__')
+    else:
+        return _is_typing_name(type(tp), name='NewType')
 
-    values = literal_values(type_)
-    return [x for value in values for x in all_literal_values(value)]
 
+def is_hashable(tp: Any, /) -> bool:
+    """Return whether the provided argument is the `Hashable` class.
 
-def is_annotated(ann_type: Any) -> bool:
-    return get_origin(ann_type) is Annotated
+    ```python test="skip" lint="skip"
+    is_hashable(Hashable)
+    #> True
+    ```
+    """
+    # `get_origin` is documented as normalizing any typing-module aliases to `collections` classes,
+    # hence the second check:
+    return tp is collections.abc.Hashable or get_origin(tp) is collections.abc.Hashable
+
+
+def is_callable(tp: Any, /) -> bool:
+    """Return whether the provided argument is a `Callable`, parametrized or not.
+
+    ```python test="skip" lint="skip"
+    is_callable(Callable[[int], str])
+    #> True
+    is_callable(typing.Callable)
+    #> True
+    is_callable(collections.abc.Callable)
+    #> True
+    ```
+    """
+    # `get_origin` is documented as normalizing any typing-module aliases to `collections` classes,
+    # hence the second check:
+    return tp is collections.abc.Callable or get_origin(tp) is collections.abc.Callable
 
 
-def annotated_type(type_: Any) -> Any | None:
-    return get_args(type_)[0] if is_annotated(type_) else None
+_PARAMSPEC_TYPES: tuple[type[typing_extensions.ParamSpec], ...] = (typing_extensions.ParamSpec,)
+if sys.version_info >= (3, 10):
+    _PARAMSPEC_TYPES = (*_PARAMSPEC_TYPES, typing.ParamSpec)  # pyright: ignore[reportAssignmentType]
 
 
-def is_unpack(type_: Any) -> bool:
-    return get_origin(type_) in UNPACK_TYPES
+def is_paramspec(tp: Any, /) -> bool:
+    """Return whether the provided argument is a `ParamSpec`.
 
+    ```python test="skip" lint="skip"
+    P = ParamSpec('P')
+    is_paramspec(P)
+    #> True
+    ```
+    """
+    return isinstance(tp, _PARAMSPEC_TYPES)
 
-def unpack_type(type_: Any) -> Any | None:
-    return get_args(type_)[0] if is_unpack(type_) else None
 
+_TYPEALIASTYPES: tuple[type[typing_extensions.TypeAliasType], ...] = (typing_extensions.TypeAliasType,)
+if sys.version_info >= (3, 12):
+    _TYPEALIASTYPES = (*_TYPEALIASTYPES, typing.TypeAliasType)
 
-def is_namedtuple(type_: type[Any]) -> bool:
-    """Check if a given class is a named tuple.
-    It can be either a `typing.NamedTuple` or `collections.namedtuple`.
-    """
-    from ._utils import lenient_issubclass
 
-    return lenient_issubclass(type_, tuple) and hasattr(type_, '_fields')
+def is_typealiastype(tp: Any, /) -> TypeIs[typing_extensions.TypeAliasType]:
```

Comment:
Same as above, would prefer `is_type_alias_type` given that it's pascal cased to imply 3 separate words